signer/src/testing/transaction_signer.rs

```diff
@@ -230,20 +254,30 @@ where
             })
             .collect();
 
-        let test_data = generate_test_data(&mut rng);
+        let test_data = self.generate_test_data(&mut rng);
         for handle in event_loop_handles.iter_mut() {
             Self::write_test_data(&test_data, &mut handle.storage).await;
         }
 
         for handle in event_loop_handles.iter() {
             handle
-                .notification_tx
+                .block_observer_notification_tx
                 .send(())
                 .expect("failed to send notification");
         }
 
-        // TODO(258): Ensure we can wait for the signers to finish processing messages
-        tokio::time::sleep(std::time::Duration::from_millis(500)).await;
+        let num_expected_decisions = (self.num_signers - 1)
```

Comment:
Why `num_signers - 1`? Is it because we are waiting for the *other* signers and this is run from the perspective of the coordinator?

Reply:
Almost, this is run from the perspective of each signer, which will receive messages from all the other signers.